src/data_sync/common.py

```diff
@@ -78,6 +78,10 @@ def compute_block_and_month_range(  # pylint: disable=too-many-locals
     current_month = (
         f"{current_month_end_datetime.year}_{current_month_end_datetime.month}"
```

Comment:
Alternatively (in a clean up):
```suggestion
        f"{current_month_end_datetime.year}_{current_month_end_datetime.month:02d}"
```
instead of padding explicitly. Or use
```python
current_month = current_month_end_datetime.strftime('%Y_%m')
```
for formatting a `datetime` object into a `YYYY_MM` representation.